assets/docs/interactivity_tutorial.md

```diff
@@ -0,0 +1,441 @@
+
+## Filters control what is shown
+Filters can be used to control what elements are shown on a panel. This will hide elements, but preserve the space that those elements would have occupied: eg, the axis limits will reflect all of the data, not just what is currently shown.
+
+Filters are specified in data layout layouts, and can be applied to most data layer types.
+```javascript
+{
+  ...options,
+  filters: [
+    { field: '{{namespace[access]}}score', operator: '!=', value: null },
+  ],
+}
+```
+
+Scatter plots have an additional option to show labels for some or all data elements. This is controlled via adding a similar filter block inside *label*:
+```javascript
+{
+  id: 'phewaspvalues',
+  ...options,
+  label: {
+    text: '{{{{namespace[phewas]}}trait_label}}',
+    filters: [
+      {
+        field: '{{namespace[phewas]}}log_pvalue',
+        operator: '>=',
+        value: 20,
+      },
+    ],
+  }
+}
+```
+
+The following filters are available:
+
+* `=`: field value *exactly* matches the provided value
+* `!=`: field value *loosely* does not match the provided value (can be used to filter out falsy values such as undefined, null, or false)
+* Numerical comparisons: `<`, `<=`,  `>`, `>=`
+* `in`: field value is present inside the provided value. The provided value can be a string, or, more commonly, an array: eg "gene type is one of the following".
+* `match`: the provided value is present in the field value. Useful for partial text search, eg "show only genes whose name contains the user-provided filter string "HLA"
+
+> *NOTE:* If a list of multiple filters is provided, then *all* filter conditions must be true for an item to be shown in the plot.
+
+Filters can be modified interactively by the user after first render: see the `filter_field` widget for details.
+
+### Imperative API (provisional)
+In the vast majority of cases, we strongly recommend using the declarative syntax for setting filters.
+
+In very rare cases, a developer may wish to control the filter logic via a bit of custom code, if the built-in filters are too restrictive.  Each data layer provides a `data_layer.setFilter(callback)` method that can be used to define the filter function explicitly. The callback will have the same arguments as the standard javascript method `Array.filter()`, and will override any declarative layout options.
+
+> *WARNING*: The imperative syntax is provisional. We reserve the right to remove or change this interface at any time, and replace it with another syntax that is more consistent with other mechanisms in LocusZoom (such as declarative filters or plugins).
+
+## Annotations preserve custom options across re-render
+LocusZoom typically maintains a separation of concerns, in which data layers are responsible for rendering the data provided by an adapter. For the most part, the display layer is not responsible for changing data, and it does not preserve state across renderings
```

Comment:
This piece is very helpful for internal use like tooltips (because the code has access to the data element directly), but it requires digging into LZ internals to use from an external widget. How can we improve usability?